gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/search/SubscriptionCriteria.java

```diff
@@ -35,7 +34,7 @@ public class SubscriptionCriteria {
 
     private final Collection<String> ids;
 
-    private final Collection<String> environments;
+    private final Set<String> environments;
```

Comment:
All others attributes are Collection, why do you know to force the type here?
I suggest to either keep it as is, or align all.

Reply:
I saw that sometimes it is a list, sometimes it is a `Set` (ex: ApiKeyCriteria) and figured out that it was particularly simpler to use set to avoid many conversions and type checks when enforcing or verifying tenant (see TenantUtils).